src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -271,15 +282,28 @@ public class TimeEntryBlock
         public ulong Started { get; set; }
         public ulong Ended { get; set; }
         public ReactiveCommand<Unit, Unit> CreateTimeEntryFromBlock { get; }
+        public ReactiveCommand<Unit,Unit> OpenEditView { get; }
+        public string TimeEntryId { get; private set; }
 
-        public TimeEntryBlock()
+        [Reactive]
+        public bool IsEditViewOpened { get; set; }
+
+        public TimeEntryBlock(string timeEntryId)
         {
+            TimeEntryId = timeEntryId;
             CreateTimeEntryFromBlock = ReactiveCommand.Create(() => AddNewTimeEntry());
+            OpenEditView = ReactiveCommand.Create(() =>
+            {
+                Toggl.Edit(TimeEntryId, false, Toggl.Description);
+            });
         }
 
+        public TimeEntryBlock() : this(null) { }
+
         private void AddNewTimeEntry()
         {
-            Toggl.CreateEmptyTimeEntry(Started, Ended);
+            TimeEntryId = Toggl.CreateEmptyTimeEntry(Started, Ended);
+            Toggl.Edit(TimeEntryId, false, Toggl.Description);

```

Comment:
I'd use `OpenEditView` command here directly, it's more readable and it will help in case the logic of `OpenEditView` changes.